plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/filtering/SemanticFilterTag.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ * 
+ * Copyright 2022 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package de.cau.cs.kieler.klighd.filtering;
+
+/**
+ * Semantic filter tags serve both as tags on graph elements to provide semantic meaning in a diagram that can not 
+ * be conveyed otherwise and as an atomic filter rule that can be combined with other rules to construct more complex
+ * rules.
+ * 
+ * @author mka
+ *
+ */
+public class SemanticFilterTag extends SemanticFilterRule {
+    
+    private String tag;
+    private Double num = 0.0;
```

Comment:
The way I understand the issue here is that elements that do not define some numerical tag, but are queried on that tag will behave as if they have set the value to 0.0, which can lead to issues if 0 is some semantically significant information for instance if negative values might be used. These quirks should be documented well here so that synthesis  writers designing tags are aware of what to consider when designing their tags and rules.